packages/eslint-plugin-gestalt/src/prefer-box-as-tag.js

```diff
@@ -0,0 +1,99 @@
+// @flow strict
+import { getHtmlTag, hasImport } from './eslintASTHelpers.js';
+import { renameTagFixer, updateGestaltImportFixer } from './eslintASTFixers.js';
+/**
+ * @fileoverview Prefer Box: prevent HTML tags supported in Box through the `as` prop
+ * @author Alberto Carreras <acarreras@pinterest.com>
+ */
+
+export const SUPPORTED_HTML_TAGS = [
+  'article',
+  'aside',
+  'details',
+  'figcaption',
+  'figure',
+  'footer',
+  'header',
+  'main',
+  'nav',
+  'section',
+  'summary',
+];
+
+const rule = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'Prefer Box: prevent HTML tags supported in Box through the `as` prop: "article" , "aside" | "details" | "div" | "figcaption" | "figure" | "footer" | "header" | "main" | "nav" | "section" | "summary", instead',
+      category: 'Gestalt alternatives',
+      recommended: true,
+      url: 'https://gestalt.pinterest.systems/Eslint%20Plugin#gestaltprefer-box-lonely-ref',
+    },
+    fixable: 'code',
+    schema: ([]: $ReadOnlyArray<empty>),
+    messages: {
+      disallowed: `Use <Box as="{{ tagName }}"></Box>.`,
+    },
+  },
+
+  // $FlowFixMe[unclear-type]
```

Comment:
@AlbertCarreras it would be great to have a `ESLintRule` flow type so we
* Ensure the type is consistent between rules
* We enforce certain properties to be set (e.g. `meta.docs.description`

```
/**
 * Flowtype definitions for eslint.ts
 * Generated by Flowgen from a Typescript Definition
 * Flowgen v1.14.1
 */

// @flow

declare type ESLintRuleMetaData = {|
  docs: {|
    /**
     * provides the short description of the rule in the [rules index](https://eslint.org/docs/rules/)
     */
    description: string,

    /**
     * specifies the heading under which the rule is listed in the [rules index](https://eslint.org/docs/rules/)
     */
    category?: string,

    /**
     * is whether the `"extends": "eslint:recommended"` property in a [configuration file](https://eslint.org/docs/user-guide/configuring#extending-configuration-files) enables the rule
     */
    recommended?: boolean,

    /**
     * specifies the URL at which the full documentation can be accessed
     */
    url?: string,

    /**
     * specifies whether rules can return suggestions (defaults to false if omitted)
     */
    suggestion?: boolean,
  |},
  messages?: {|
    [messageId: string]: string,
  |},
  fixable?: 'code' | 'whitespace',
  schema?: any,
  deprecated?: boolean,
  type?: 'problem' | 'suggestion' | 'layout',
|};

declare interface ESLintRule {
  create(context: any): any;
  meta?: ESLintRuleMetaData;
}
```

Reply:
Thanks! DONE